release/models/platform/openconfig-platform.yang

```diff
@@ -531,6 +539,27 @@ module openconfig-platform {
         this reports the role of the component.";
     }
 
+    container last-poweroff-reason {
+      description
+        "Records last power-off reason for a component.";
+
+      uses oc-platform-types:component-last-poweroff-reason;
+    }
+
+    leaf last-poweroff-time {
+      type oc-types:timeticks64;
+      units "nanoseconds";
+      description
+        "This records the last time a component was directly powered
+         down. The value is a Unix Epoch timestamp (nanoseconds since
+         Jan 1, 1970 00:00:00 UTC). Component power-off can be:
+         - USER_INITIATED
+         - SYSTEM_INITIATED
+         - POWER_FAILURE
```

Comment:
If `power_failure` is not a controlled shutdown (e.g. in a system with many PSUs, some of the PSUs fail and system initiates a proactive, graceful shutdown of the component while there's still enough power to keep the control modules up), then it will not be possible to record the `power_failure` event. 

In other words, it is possible to keep track of only the controller shutdowns

Reply:
Sure, not a immediate requirement for support atm but I would like to capture it here.